eth/eth.go

```diff
@@ -88,3 +92,23 @@ func (e *Block) toRPCBlock(block *monomer.Block, fullTx bool) (map[string]any, e
 	}
 	return rpcBlock, nil
 }
+
+type ProofProvider struct {
+	blockchainAPI *ethapi.BlockChainAPI
+}
+
+func NewProofProvider(db state.Database, blockStore DB) *ProofProvider {
+	return &ProofProvider{
+		blockchainAPI: ethapi.NewBlockChainAPI(newEthAPIBackend(db, blockStore)),
+	}
+}
+
+// GetProof returns the account and storage values of the specified account including the Merkle-proof.
+// The user can specify either a block number or a block hash to build to proof from.
+func (p *ProofProvider) GetProof(
+	address common.Address,
```

Comment:
We should accept a `context.Context` here as well (it is permissible with the geth RPC server).

Reply:
Good catch, I'll add that in the first commit and then merge